std/meta.d

```diff
@@ -1313,6 +1314,45 @@ private template SmartAlias(T...)
 }
 
 /**
+  * This template instantiates a template by taking its parameters first, and
+  * the template last. This can be useful when using staticMap to apply various
+  * templates to the same set of parameters, since staticMap always appends the
+  * list of arguments to one template.
+  *
+  * `ApplyWith!(A0, A1, ..., An-1, An)` is equivalent to
+  * `An!(A0, A1, ..., An-1)`
```

Comment:
I kept looking at this and I might be wrong, but I can't stop wondering if it wouldn't be more natural for the call to be `A0!(A1, ..., An)` ?
If so, maybe we could have the template declaration as `ApplyWith(Func, Args...)` ?
What do you think?

Reply:
I get what you are saying. By itself, `ApplyWith` (or whatever we should call it) looks weird when you add the template at the end. The whole point of this template is to work with `ApplyLeft`. It's like specifying all the parameters, and leaving the actual template to be used up to a later time.

For example:

```D
template T1(T) {}
template T2(T) {}

alias X = ApplyLeft!(ApplyWith!string);

alias a = X!T1; // equivalent to T1!string
alias b = X!T2; // equivalent to T2!string
```

I'm thinking now, since pretty much `ApplyLeft` is required to make any sense, `ApplyWith` should use it internally instead of `Instantiate`. The fact that it's used can be hidden.